stdlib/public/TensorFlow/TensorFlow+NumPy.swift

```diff
@@ -0,0 +1,68 @@
+//===-- TensorFlow+NumPy.swift --------------------------------*- swift -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines bridging from `numpy.ndarray` to `ShapedArray` and
+// `Tensor`.
+//
+//===----------------------------------------------------------------------===//
+
+import Python
+
+public extension ShapedArray {
+  init?(numpyArray: PyValue) {
+    guard let np = try? Python.attemptImport("numpy") else {
+      return nil
+    }
+    // Check if input is a `numpy.ndarray` instance.
+    guard Python.isinstance.call(with: numpyArray, np.ndarray) == true else {
+      return nil
+    }
+
+    // Check if the array's dtype matches the `Scalar` type.
+    // (e.g. `np.float32` and `Float`).
+    var scalarTypeMatch = false
+    switch numpyArray.dtype {
```

Comment:
Use a `let` here and initialize it entirely in the `switch` statement. Assign `false` to it in `default:`.